src/core/search/parser.y

```diff
@@ -178,6 +178,7 @@ tag_list_element:
   TERM { $$ = std::move($1); }
```

Comment:
I do you need both TERM and TAG_VAL here?

Reply:
yes, because the term can be a tag or another part of the query

Reply:
Can you please provide an example of a tag_list_element that matches TERM but not TAG_VAL?

Reply:
Ok, I remember now that lexer parses first tokens and then grammar matches them